dart/lib/src/scope.dart

```diff
@@ -142,6 +142,65 @@ class Scope {
   /// Removes an extra from the Scope
   void removeExtra(String key) => _extra.remove(key);
 
+  SentryEvent applyToEvent(SentryEvent event, dynamic hint) {
+    if (event.transaction == null) {
+      event = event.copyWith(transaction: transaction);
+    }
```

Comment:
what's about testing if the scope values are not null either? so we avoid extra alocation if theres nothing to be copied

eg:
if (event.transaction == null && transaction != null) {
      event = event.copyWith(transaction: transaction);
}

Reply:
should we set the breadcrumbs if the list is empty ?
also what about the same syntax than _prepareEvent : 
```dart
SentryEvent applyToEvent(SentryEvent event, dynamic hint) {
  return event.copyWith(
    transaction: event.transaction ?? transaction,
    user: event.user ?? user,
    // ...
```